src/main/java/org/cirdles/topsoil/app/TopsoilMainWindow.java

```diff
@@ -104,10 +105,25 @@ private void initialize() {
                 .forEach(this::loadDataset);
 
         reloadDatasetMenu();
+
+        if (datasetManager.getDatasets().isEmpty()) {
+            createInstructionsTab();
+        }
+    }
+
+    Optional<TabContents> getCurrentTabContents() {
+        if (!getCurrentTab().isPresent()) {
+            //No tab
+            warningPrompt("There's no table open.");
+            return Optional.empty();
+
+        } else {
+            return getCurrentTab().map(tab -> (TabContents) tab.getContent());
+        }
     }
 
     Optional<TSVTable> getCurrentTable() {
-        return getCurrentTab().map(tab -> (TSVTable) tab.getContent());
+        return getCurrentTabContents().map(table -> table.getTable().orElse(null));
```

Comment:
This is not good. Please look at how we've used `Optional` elsewhere or maybe check for resources online. Hint: Use `flatMap`.